### DISCUSSION TITLE
How to disable warning prints?

### DISCUSSION THREAD

**Discussion by denniskb** [alandefreitas/matplotplusplus#247] [Q&A]
Some of my matplotplusplus calls generate warnings messages that are printed to `stdout`. Things such as "warning: empty scatter plot". Since I'd like to print to `stdout` myself this is rather unfortunate. Is there a way to disable all warning prints or perhaps redirect them to a different stream (i.e. `/dev/null`)? Many thanks!

**Comment by alandefreitas** [comment]
This is related to how the pipes are created in [`gnuplot.h`](https://github.com/alandefreitas/matplotplusplus/blob/master/source/matplot/backend/gnuplot.h) and [`gnuplot.cpp`](https://github.com/alandefreitas/matplotplusplus/blob/master/source/matplot/backend/gnuplot.cpp). 

It's not hard to reimplement this logic to redirect the messages differently or even copy and paste these files to create another backend locally that would handle the pipes differently, like throwing an exception instead or something like that. I think this request has never come up because only errors are redirected to `stdout` and the easiest solution is just to fix the error that is leading to the warning.

**Reply by denniskb** [reply]
I could skip calling `scatter()` with empty `x` and `y` but then the colors of subsequent plots change. Is there a way to "advance the color palette" without issuing a plot call?

**Reply by alandefreitas** [reply]
The simplest solution is to define explicit colors instead of relying on automatic colors. You can even reuse the automatic colors if you want:

https://github.com/alandefreitas/matplotplusplus/blob/280659aa6835acdb16983e2af4fa704bca98a503/source/matplot/core/axes_type.h#L2832

If you want to rely on the automatic colors from the `axes_type`, you can call 

https://github.com/alandefreitas/matplotplusplus/blob/280659aa6835acdb16983e2af4fa704bca98a503/source/matplot/core/axes_type.h#L495

On the current axes object and ignore the returned value.

This is what the plot objects do under the hood:

https://github.com/alandefreitas/matplotplusplus/blob/280659aa6835acdb16983e2af4fa704bca98a503/source/matplot/axes_objects/line.cpp#L229
